contracts/test/WhitelistPlugin.spec.ts

```diff
@@ -0,0 +1,146 @@
+import hre, { deployments } from "hardhat";
+import { expect } from "chai";
+import { SignerWithAddress } from "@nomicfoundation/hardhat-ethers/signers";
+import { getWhiteListPlugin, getRegistry, getManager } from "../src/utils/contracts";
+import { loadPluginMetadata } from "../src/utils/metadata";
+import { IntegrationType } from "../src/utils/constants";
+import { buildSingleTx } from "../src/utils/builder";
+
+describe("WhitelistPlugin", async () => {
+    let deployer: SignerWithAddress, owner: SignerWithAddress, user1: SignerWithAddress, user2: SignerWithAddress, user3: SignerWithAddress;
+
+    before(async () => {
+        [deployer, owner, user1, user2, user3] = await hre.ethers.getSigners();
+    });
+
+    const setup = deployments.createFixture(async ({ deployments }) => {
+        await deployments.fixture();
+        const plugin = await getWhiteListPlugin(hre);
+        const registry = await getRegistry(hre, owner.address);
+        const safe = await (await hre.ethers.getContractFactory("TestExecutor")).deploy();
+        const manager = await getManager(hre, owner.address, await registry.getAddress());
+
+        registry.addIntegration(await plugin.getAddress(), IntegrationType.Plugin);
+
+        safe.setModule(await manager.getAddress());
+
+        const data = manager.interface.encodeFunctionData("enablePlugin", [await plugin.getAddress(), false]);
```

Comment:
For testing the plugin this should not be required. The plugin interacts with the manager and doesn't care about the rest. For this we should use a mock (that is why it was setup to return a mock for tests). This way you can test how it should behave in different cases (i.e. in case the manager reverts).